source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -482,12 +508,157 @@ Pass::Status AggressiveDCEPass::ProcessImpl() {
   // If any extensions in the module are not explicitly supported,
   // return unmodified.
   if (!AllExtensionsSupported()) return Status::SuccessWithoutChange;
-  // Process all entry point functions
+
+  // Eliminate Dead functions.
+  bool modified = EliminateDeadFunctions();
+
+  InitializeModuleScopeLiveInstructions();
+
+  // Process all entry point functions.
   ProcessFunction pfn = [this](ir::Function* fp) { return AggressiveDCE(fp); };
-  bool modified = ProcessEntryPointCallTree(pfn, get_module());
+  modified |= ProcessEntryPointCallTree(pfn, get_module());
+
+  // Process module-level instructions.
+  modified |= ProcessGlobalValues();
```

Comment:
Should this be done before AggressiveDCE()?  That may generate more dead global values.

Reply:
No, the way pass works most global values would look dead. They need to be marked live through uses in the functions first.